android/src/main/AndroidManifest.xml

```diff
@@ -60,6 +60,7 @@
 			<intent-filter>
 				<action android:name="com.tailscale.ipn.CONNECT_VPN" />
 				<action android:name="com.tailscale.ipn.DISCONNECT_VPN" />
+				<action android:name="com.tailscale.ipn.INIT_WITH_AUTHKEY" />
```

Comment:
If I'm reading this correctly, any app can send a disconnect broadcast, then this and cause the device to join a particular tailnet. We need to make sure this intent is not callable or receivable by other applications.

Reply:
Would this be `len(b.backend.ListProfiles()) == 0` ? If any authenticated profiles exist, whether actively logged in or not, then we wouldn't accept an authkey?

Reply:
Yeah I didn't finish this part.